src/compute/src/compute_state.rs

```diff
@@ -176,6 +187,40 @@ impl ComputeState {
             .get_mut(&id)
             .expect("collection must exist")
     }
+
+    /// Apply the current `worker_config` to the compute state.
+    fn apply_worker_config(&mut self) {
```

Comment:
Note to self: We should have a mechanism to apply per-process configs. All configs mentioned in the current implementation apply process-wide, and aren't scoped to the calling thread.

Reply:
Yep, I missed that it changes the join spec, too. That certainly needs to be applied per-thread.